earthpy/spatial.py

```diff
@@ -121,8 +121,138 @@ def normalized_diff(b1, b2):
+         
+    with contextlib.ExitStack() as context:
+        sources = [
+            context.enter_context(rio.open(path, **kwds))
+            for path in band_paths
+        ]
+
+        # TODO: Check that the CRS and TRANSFORM are the same
+        dest_kwargs = sources[0].meta
+        dest_count = sum(src.count for src in sources)
+        dest_kwargs["count"] = dest_count
+
+        # Stack the bands and return an array, but don't write to disk
+        if not write_raster: 
+            
+            arr, prof = _stack_bands(sources)
+            return arr, prof
+        
+        # Write out the stacked array and return a numpy array    
+        else:
+            # Valid output path checked above
+            file_fmt = os.path.basename(out_path).split('.')[-1]
+            
+            # Check if the file format for output is the same as the source driver
+            rio_driver = sources[0].profile['driver']
+            if not file_fmt in rio_driver.lower():
+                raise ValueError("Source data is {}. Please specify corresponding output extension.".format(rio_driver))
+            
+            # Write stacked gtif file
+            with rio.open(out_path, "w", **dest_kwargs) as dest:
+                _stack_bands(sources, dest, write_raster)
+            
+            # Read and return array
+            with rio.open(out_path, "r") as src:
+                return src.read(), src.profile
+        
+                
+def _stack_bands(sources, dest=None, write_raster=False):
+    """Stack a set of bands into a single file.
+
+    Parameters
+    ----------
+    sources : list of rasterio dataset objects
+        A list with paths to the bands you wish to stack. Objects
+        will be stacked in the order provided in this list.
+    dest : a rio.open writable object that will store raster data.
+    write_raster: a flag to decide to write out the raster. 
+    """
+
+    try:
+        for src in sources:
+            src.profile
+
+    except ValueError as ve:
+        raise ValueError("The sources object should be Dataset Reader")
+        sys.exit()
+
+    else:
+        pass
+
+    if write_raster:
+        for ii, ifile in enumerate(sources):
+            bands = sources[ii].read()
+            if bands.ndim != 3:
+                bands = bands[np.newaxis, ...]
+            for band in bands:
+                dest.write(band, ii + 1)
+                
+    else:
+        stacked_arr = []
+        for ii, ifile in enumerate(sources):
+            bands = sources[ii].read()
+            if bands.shape[0] == 1:
+                bands = np.squeeze(bands)
+            stacked_arr.append(bands)
+        
+        # update the profile to have count==number of bands
+        ret_prof = sources[0].profile.copy()
+        ret_prof['count'] = len(stacked_arr)
+        
+        return np.array(stacked_arr), ret_prof
+
+# @deprecate
```

Comment:
Why don't we move this deprecated function to the bottom of this file. @mbjoseph do you have any thoughts on the location of deprecated functions within our package??? i was just thinking moving it down but am open to suggestions

Reply:
I think the `@deprecate` decorator is probably fine, but could see a justification for using that *and* moving deprecated things to the bottom of the module.

Reply:
ok great. Thank you @mbjoseph  !!!